modules/ROOT/pages/understanding-orchestration-using-mule.adoc

```diff
@@ -0,0 +1,76 @@
+
+== When to Use Orchestration
+
+Use orchestration when you need to combine existing services to meet business requirements. For example, you might use orchestration when you're adding a new employee who needs to be entered in both the payroll system and HR system. You might also use orchestration for order processing. In this scenario, you can use orchestration to accept an order, make sure the order is in the inventory system., use a credit check system to ensure that you trust the customer's credit, and enter the order into billing and shipping systems.
+
+== An Example of Orchestration
+
+The following example of orchestration uses a flow to do the following:
+
+. Transform the file’s contents into a standard format that contains one or more specific orders.
+. Split out the orders for separate processing and filter out non-book orders.
+. Call two services: one to check whether the book is in the inventory, and the other to create a customer order.
+. Email the order to the customer and store it in a database.
+
+If any step in this process fails, a message is placed in a JMS queue of failed orders. You can view, fix, and resubmit the affected orders.
+
+[source,xml,linenums]
+----
+<flow name="orderHandling">
+  <file:listener directory="/myDirectory">
+    <scheduling-strategy>
+      <fixed-frequency/>
+    </scheduling-strategy>
+    <file:matcher filenamePattern="*.xml"/>
+  </file:listener>
+  <ee:transform>
+    <ee:message>
+      <ee:set-payload resource="createBookOrdersTransformation.dwl"/>
+    </ee:message>
+  </ee:transform>
+  <foreach collection="#[payload.orders filterObject ((value, key, index) -> value.type == 'books')]">
+      <http:request method="POST" path="/book/inventory"/>
+      <http:request method="POST" path="/book/order"/>
+      <email:send toAddresses="#[payload.customer.email]">
+        <email:body >
+          <email:content ><![CDATA[#['Your order has been placed.']]]></email:content>
+        </email:body>
+      </email:send>
+      <db:insert>
+        <db:sql >INSERT INTO ORDERS(CUSTOMER, AMOUNT) VALUES (:id, :amount)</db:sql>
+        <db:input-parameters><![CDATA[#[{ id : payload.customer.id, amount: payload.order.total }]]]></db:input-parameters>
+      </db:insert>
+    </choice>
+  </foreach>
+  <error-handler >
+    <on-error-continue>
+      <jms:publish destination="failedOrders"/>
+    </on-error-continue>
+  </error-handler>
+</flow>
+----
+
+== See Also
+
+// * xref:using-flows-for-service-orchestration.adoc[Using Flows for Service Orchestration]
```

Comment:
I assume this link will be also merged, since 4.1 version is already available. 
https://docs.mulesoft.com/mule-runtime/4.1/using-flows-for-service-orchestration

Reply:
Once that gets into 4.2 I can uncomment the link to make it live.